src/paho/mqtt/publish.py

```diff
@@ -18,15 +18,49 @@
 situation where you have a single/multiple messages you want to publish to a
 broker, then disconnect and nothing else is required.
 """
+from __future__ import annotations
 
 import collections
+import typing
 from collections.abc import Iterable
+from typing import Any, List, Tuple, Union
```

Comment:
Might be good to import the `typing` one way only; I'd suggest the `from typing` way.